sdk/trace/provider.go

```diff
@@ -263,7 +263,9 @@ func WithSpanProcessor(sp SpanProcessor) TracerProviderOption {
 func WithResource(r *resource.Resource) TracerProviderOption {
 	return func(opts *TracerProviderConfig) {
 		if r != nil {
```

Comment:
No need for the conditional here, `resource.Merge(resource.Environment(), r)` will handle `r` being `nil`, similar to what was done in the metrics `Controller`.